packages/server/src/modules/auth/auth.resolver.ts

```diff
@@ -180,20 +182,72 @@ export class AuthResolver {
     });
   }
 
-  @Mutation(() => GraphQLVoid, { nullable: true })
-  async sendRegistrationInvite(
+  @Mutation(() => GraphQLUser, { nullable: true })
+  async addOrInviteOperator(
     @Input() { email, retailLocationId }: RegistrationInviteLinkPayload,
-    @CurrentUser() operator: User,
+    @CurrentUser() currentUser: User,
   ) {
+    await this.authService.assertMembership({
+      userId: currentUser.id,
+      message: "You are not allowed to add a new Operator",
+      role: "ADMIN",
+      retailLocationId,
+    });
+
+    const userToUpdate = await this.userService.findUserByEmail(email);
+
+    if (userToUpdate) {
+      const currentOperatorMembership =
+        await this.prisma.locationMember.findUnique({
+          where: {
+            userId_retailLocationId: {
+              userId: userToUpdate.id,
+              retailLocationId,
+            },
+          },
+        });
+
+      if (currentOperatorMembership?.role) {
+        if (currentOperatorMembership.role === "OPERATOR") {
+          throw new UnprocessableEntityException(
+            "You cannot promote to operator someone who already is an operator in this location.",
+          );
+        } else {
+          throw new UnprocessableEntityException(
+            "The selected user is already an admin.",
+          );
+        }
+      }
+
+      if (!userToUpdate.emailVerified) {
+        throw new UnprocessableEntityException(
+          "The selected user needs to verify their email first.",
+        );
+      }
+
+      return (
+        await this.prisma.locationMember.create({
+          data: {
+            role: "OPERATOR",
+            retailLocationId,
+            userId: userToUpdate.id,
+          },
+          include: {
+            user: true,
+          },
+        })
+      ).user;
```

Comment:
The user should stay the same, so no need for an extra query:
```suggestion
      await this.prisma.locationMember.create({
        data: {
          role: "OPERATOR",
          retailLocationId,
          userId: userToUpdate.id,
        },
      });
      return userToUpdate;
```
or, if there is a need I can't see now:
```suggestion
      const { user } = await this.prisma.locationMember.create({
        data: {
          role: "OPERATOR",
          retailLocationId,
          userId: userToUpdate.id,
        },
        select: { user: true },
      });
      return user;
```